client/src/entrypoints/admin/bulk-actions.js

```diff
@@ -146,6 +181,9 @@ function onSelectIndividualCheckbox(e) {
     document.querySelector(BULK_ACTION_NUM_OBJECTS).textContent =
       numObjectsSelected;
   }
+
+  // Updating previously checked object
+  checkedState.prevCheckedObject = Number(e.target.dataset.objectId);
```

Comment:
What if the `objectId` is not a number value?

Reply:
Actually I assumed that it will be a number because I saw that `Number(e.target.dataset.objectId)` was used before for adding checkbox items to `checkedState.checkedObjects`

Reply:
I have validated that this breaks if you do not have a number as a model id and raised a bug for this - https://github.com/wagtail/wagtail/issues/8563

Reply:
I still think we can store a reference to the object id, just avoid converting it to a number. 

Using a reference to another Dom element is more risky, if any code introduces something like drag & drop or other Dom manipulation the reference may become stale. 

However, if you find you have to convert to a Number for it to work, we can solve that issue separately. 

See how you go.

Reply:
Sure! I will try implementing it without converting it to number and see if it works. I will soon make a commit for it.

Reply:
I have made a commit for this. I just avoided converting it to number and it still worked. Please have a look.